lib/floe/workflow/states/wait.rb

```diff
@@ -28,17 +28,21 @@ def initialize(workflow, name, payload)
 
         def start(input)
           super
-          input = input_path.value(context, input)
 
-          context.output     = output_path.value(context, input)
-          context.next_state = end? ? nil : @next
+          input          = input_path.value(context, input)
+          context.output = output_path.value(context, input)
 
           wait_until!(
             :seconds => seconds_path ? seconds_path.value(context, input).to_i : seconds,
             :time    => timestamp_path ? timestamp_path.value(context, input) : timestamp
           )
         end
 
+        def finish
+          context.next_state = end? ? nil : @next
```

Comment:
Wouldn't this be the same for all states?  I'm curious why wait is different.  I woiuld have expected that any state would describe the next state immediately (except for possibly choice)

Reply:
Yeah true, not all states have Next but I could move this to the NonTerminalMixin